### PR TITLE
Add `merge_options` for `haproxy::defaults`

### PR DESCRIPTION
**Pull request by yakatz**
## Summary
There are default options which would be nice to have in all defaults sections. Add an option to include them.

## Checklist
- [x] 🟢 Spec tests.
- [x] 🟢 Acceptance tests.
- [x] Manually verified. (For example `puppet apply`)

### CODE REVIEW

**Review by Ramesh7 — APPROVED** [review]

**Comment by Ramesh7** [comment]
Changes looks good to me but I am not expert on this module. If we can get another review from community, we should be good to go after that.

**Review by bastelfreak — DISMISSED** [review]

File: manifests/defaults.pp
Comment (bastelfreak) [inline manifests/defaults.pp:51]: would be nice if we could do the major release soonish to flip the default value.

Comment (smortex) [reply manifests/defaults.pp:51]: Fun thing: we have the same message (permalink: https://github.com/puppetlabs/puppetlabs-haproxy/blame/a3d28a151723b059c4689d628d60b93b7388c2a5/manifests/config.pp#L28) added in 8adbf88bc353c3882e9c3a5a3b92e20ffa830f26, first released in version 1.4.0 and still here in 7.2.1

Comment (bastelfreak) [reply manifests/defaults.pp:51]: Instead of adding another warning, we should flip it: https://github.com/puppetlabs/puppetlabs-haproxy/pull/592

Comment (bastelfreak) [reply manifests/defaults.pp:51]: I suggest to merge #592, then update #588 and remove the warning.

Comment (bastelfreak) [reply manifests/defaults.pp:51]: @yakatz can you please rebase against main and remove the warning? #592 was merged

**Review by bastelfreak — APPROVED** [review]

**Review by Ramesh7 — APPROVED** [review]

**Pull request merged**
